### PR TITLE
Add git

### PR DESCRIPTION
**Pull request by jakirkham**
Basically just dropped in the conda-recipes git with a version bump and minor tweaks.

### CODE REVIEW

**Comment by jakirkham** [comment]
@ilanschnell @frol @ccordoba12 @msarahan would be great to have your feedback here as you have been working on this for awhile. Also, if any of you would be willing to be added as maintainers, that would be great. Thanks.

**Comment by jakirkham** [comment]
Given the complexity of git, I am relieved that it went through without a hitch. Now to work on some more tests.

File: recipes/git/bld.bat [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+del post-install.bat
+
+IF NOT EXIST %PREFIX%\Menu mkdir %PREFIX%\Menu
+copy %RECIPE_DIR%\menu-windows.json %PREFIX%\Menu\
+copy %RECIPE_DIR%\git-for-windows.ico %PREFIX%\Menu\
+
+echo export PATH=$(cygpath -a %PREFIX:\=/%)/Library/bin:$PATH >> %LIBRARY_PREFIX%\etc\profile.d\env.sh
+echo. >> %LIBRARY_PREFIX%\etc\profile.d\env.sh
+
+exit 0
```
Comment (ocefpaf) [inline recipes/git/bld.bat]: I guess that building from source is not an option for now, right?
I see that you are removing `git_bash.exe`. What is the reason for that? I thought that git needs git-bash to work properly on Windows. (Even if that is not true I am interested into having git bash there for teaching purposes :wink:)

Comment (jakirkham) [inline recipes/git/bld.bat]: I blatantly lifted this from conda-recipes. In this case, @msarahan would be the one to ask.

Comment (msarahan) [inline recipes/git/bld.bat]: AFAICT building from source will become an option when @mingwandroid gets a bit further on packaging MSYS2 inside conda.  I don't know if anyone has succeeded at building git with a native Windows toolchain (I couldn't find any examples).

The EXEs that are removed are ones that live in the root of that extracted package.  There are equivalents in bin that are not deleted.  There's both git-bash and bash (which I think do the same thing?)

I have used an earlier version of this recipe to teach a SWC class.  It worked very well - especially with the shortcut that this recipe installs.  It actually only had problems on Mac (due to certificates not being found.)  I did fix that - I'll be sure to merge those changes here.  The fix was basically to download certs from cURL's site and put them in the correct place, then tell git where to find them.

Comment (ocefpaf) [inline recipes/git/bld.bat]: > I have used an earlier version of this recipe to teach a SWC class

Aha! You answered my question. Thanks!

**Comment by jakirkham** [comment]
Given this is working, I am ok with it being merged. I can always add more tests on the feedstock.

**Comment by jakirkham** [comment]
@ocefpaf, do you feel like this is ready or do you have more questions?

**Comment by ocefpaf** [comment]
I would love to see everything built from source I understand that

    Although practicality beats purity.

:+1:

**Comment by jakirkham** [comment]
Thanks @ocefpaf.

File: recipes/git/bld.bat (hunk 1 above)
Comment (ccordoba12) [inline recipes/git/bld.bat]: I'd also like to add that I almost always use git inside cmd.exe, and haven't encountered any problem so far :-)

Comment (ocefpaf) [inline recipes/git/bld.bat]: Good to know. Thanks!

Comment (msarahan) [inline recipes/git/bld.bat]: Tab completion works for branch names in git bash but not in cmd using bash.  I also use them both, though.

**Comment by pelson** [comment]
*Way* cool. :+1: 

Thanks for the pragmatic merge @ocefpaf. I completely agree with your sentiment.

**Pull request merged**